docs/source/CVAOpt_Algorithm/Parameters_and_Hyperparameters.md

```diff
@@ -82,4 +89,89 @@ with TQ42Client() as client:
+from tq42.project import list_all as list_all_projects
+from tq42.experiment import list_all as list_all_experiments
+from tq42.experiment_run import ExperimentRunStatusProto
+from tq42.algorithm import AlgorithmProto
+from tq42.compute import HardwareProto
+import json
+import pandas
+import asyncio
+import numpy as np
+from tq42.channel import Channel, Ask, Tell
+import OptimizationTestFunctions as otf
+
+cva_params = {}
+cva_params['objectives'] = [{'name': 'Rosenbrock', 'aim_type':'MINIMIZE'}]
+cva_params['variables'] = []
+cva_params['variables'].append({'name': 'x1', 'info_real':{'lower_bound':-1.0, 'upper_bound':1.0}})
+cva_params['variables'].append({'name': 'x2', 'info_real':{'lower_bound':-1.0, 'upper_bound':1.0}})
+cva_params['parameters'] = {}
+cva_params['parameters']['max_generation'] = 50
+cva_params['parameters']['mue'] = 2
+cva_params['parameters']['lambda'] = 10
+
+async def run_exp_with_channel(client, experiment_id, cva_params):
+    # set up channel
+    channel = await Channel.create(client=client)
+    # extend cva_params with func_eval_worker_channel_id
+    cva_params['func_eval_worker_channel_id'] = channel.id
+    
+    # create the experiment run
+    run = ExperimentRun.create(
+        client=client, 
+        algorithm=AlgorithmProto.CVA_OPT, 
+        experiment_id="your_experiment_id",
+        compute=HardwareProto.SMALL,
+        parameters={'parameters': cva_params, 'inputs': {} }
+    )
+    print(f\" starting run with id {run.id}\")
+
+    # define the callback function
+    async def callback(ask: Ask) -> Tell:
+        dim = len(ask.headers)
+        func = otf.Rosenbrock(dim)
+        print('callback received parameters of length: ' + str(len(ask.parameters)))
+        y = []
+        for parameter in ask.parameters:
+            y.append(float(func(np.array(parameter.values))))
+        # add result to data
+
+        return Tell(
+            parameters=ask.parameters,
+            headers=ask.headers,
+            results=y
+        )
+    # define a function to be called after the optimization is finished
+    def success():
+        pass
+
+    # let the channel wait for connections
+    is_finished = False
+    max_retries = 10
+    retries = 0
+    while (not is_finished) and (retries < max_retries):
+        await channel.connect(
+                callback=callback,
+                finish_callback=success,
+                max_duration_in_sec=None,
+                message_timeout_in_sec=120
+            )
+        retries += 1
+        run_state = run.check().data.status
+        if ExperimentRunStatusProto.Name(run_state) in ['CANCELLED', 'COMPLETED', 'FAILED', 'CANCEL_PENDING']:
+            is_finished = True
+        else:
+            print('run is in state ' + ExperimentRunStatusProto.Name(run_state) + ', continuing')
+    # return the run to retrieve the result    
+    return run
+
+run = await run_exp_with_channel(tq42client, experiment_id, cva_params)
```

Comment:
this does not work in a normal script because await is not by default enabled, you need to run inside an event loop (only jupyter notebooks have this by default).